File: CS/Grid_BottomNewItemRow/BottomNewItemRowEx.cs
using DevExpress.Skins;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Grid_BottomNewItemRow
{
    public class BottomNewItemRowEx
    {
        private const int buffer = 4;
        public BottomNewItemRowEx(GridSplitContainerDescendant splitContainer)
        {
            _Container = splitContainer;
            CustomizeSplitContainer();
        }

        
        //Fileds...
        GridSplitContainerDescendant _Container;
        GridViewDescendant firstView;
        GridViewDescendant secondView;
        GridViewInfoDescendant firstVI;
        GridViewInfoDescendant secondVI;

        private void CustomizeSplitContainer()
        {
            _Container.SizeChanged += _Container_SizeChanged;

            CustomizeFirstView();
            _Container.ShowSplitView();
            CustomizeSecondView();            

            SetSplitterPosition();

        }

        private void CustomizeFirstView()
        {
            firstView = _Container.Grid.MainView as GridViewDescendant;
            firstView.OptionsView.RowAutoHeight = true;
            firstVI = firstView.GetViewInfo() as GridViewInfoDescendant;

            firstView.OptionsView.NewItemRowPosition = NewItemRowPosition.None;
            firstView.OptionsMenu.ShowSplitItem = false;
            firstView.ScrollEvent += firstView_ScrollEvent;
 
        }

        private void CustomizeSecondView()
        {
            secondView = _Container.SplitChildView as GridViewDescendant;
            secondVI = secondView.GetViewInfo() as GridViewInfoDescendant;

            secondView.OptionsView.NewItemRowPosition = NewItemRowPosition.Bottom;
            secondView.CustomRowFilter += secondView_CustomRowFilter;
            secondView.VertScrollVisibility = DevExpress.XtraGrid.Views.Base.ScrollVisibility.Never;
 
        }

        private int i = 0;

        public void SetSplitterPosition()
        {
            Skin skin = CommonSkins.GetSkin(_Container.Grid.LookAndFeel);
            SkinElement element = skin[CommonSkins.SkinSplitterHorz];

            int newSplitterPosition = _Container.ClientSize.Height - secondVI.ViewRects.RowsTotalHeight - buffer - element.Size.MinSize.Height;
          
            if (newSplitterPosition <= 1)
                return;

            _Container.SplitterPosition = newSplitterPosition;
         
        }


        void _Container_SizeChanged(object sender, EventArgs e)
        {
            SetSplitterPosition();
        }

        private void firstView_ScrollEvent(object sender, EventArgs e)
        {
            secondVI.secondViewVScroll = (firstVI.VScrollBarPresence == ScrollBarPresence.Visible) ? ScrollBarPresence.Unknown : ScrollBarPresence.Hidden;

            secondView.LayoutChanged();
        }

        private void secondView_CustomRowFilter(object sender, DevExpress.XtraGrid.Views.Base.RowFilterEventArgs e)
        {
            e.Visible = false;
            e.Handled = true;
        }

    }
}

File: VB/Grid_BottomNewItemRow/BottomNewItemRowEx.vb
Imports DevExpress.Skins
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace Grid_BottomNewItemRow
	Public Class BottomNewItemRowEx
		Private Const buffer As Integer = 4
		Public Sub New(ByVal splitContainer As GridSplitContainerDescendant)
			_Container = splitContainer
			CustomizeSplitContainer()
		End Sub


		'Fileds...
		Private _Container As GridSplitContainerDescendant
		Private firstView As GridViewDescendant
		Private secondView As GridViewDescendant
		Private firstVI As GridViewInfoDescendant
		Private secondVI As GridViewInfoDescendant

		Private Sub CustomizeSplitContainer()
			AddHandler _Container.SizeChanged, AddressOf _Container_SizeChanged

			CustomizeFirstView()
			_Container.ShowSplitView()
			CustomizeSecondView()

			SetSplitterPosition()

		End Sub

		Private Sub CustomizeFirstView()
			firstView = TryCast(_Container.Grid.MainView, GridViewDescendant)
			firstView.OptionsView.RowAutoHeight = True
			firstVI = TryCast(firstView.GetViewInfo(), GridViewInfoDescendant)

			firstView.OptionsView.NewItemRowPosition = NewItemRowPosition.None
			firstView.OptionsMenu.ShowSplitItem = False
			AddHandler firstView.ScrollEvent, AddressOf firstView_ScrollEvent

		End Sub

		Private Sub CustomizeSecondView()
			secondView = TryCast(_Container.SplitChildView, GridViewDescendant)
			secondVI = TryCast(secondView.GetViewInfo(), GridViewInfoDescendant)

			secondView.OptionsView.NewItemRowPosition = NewItemRowPosition.Bottom
			AddHandler secondView.CustomRowFilter, AddressOf secondView_CustomRowFilter
			secondView.VertScrollVisibility = DevExpress.XtraGrid.Views.Base.ScrollVisibility.Never

		End Sub

		Private i As Integer = 0

		Public Sub SetSplitterPosition()
			Dim skin As Skin = CommonSkins.GetSkin(_Container.Grid.LookAndFeel)
			Dim element As SkinElement = skin(CommonSkins.SkinSplitterHorz)

			Dim newSplitterPosition As Integer = _Container.ClientSize.Height - secondVI.ViewRects.RowsTotalHeight - buffer - element.Size.MinSize.Height

			If newSplitterPosition <= 1 Then
				Return
			End If

			_Container.SplitterPosition = newSplitterPosition

		End Sub


		Private Sub _Container_SizeChanged(ByVal sender As Object, ByVal e As EventArgs)
			SetSplitterPosition()
		End Sub

		Private Sub firstView_ScrollEvent(ByVal sender As Object, ByVal e As EventArgs)
            secondVI.secondViewVScroll = If(firstVI.VScrollBarPresence = ScrollBarPresence.Visible, ScrollBarPresence.Unknown, ScrollBarPresence.Hidden)

            secondView.LayoutChanged()
		End Sub

		Private Sub secondView_CustomRowFilter(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.RowFilterEventArgs)
			e.Visible = False
			e.Handled = True
		End Sub

	End Class
End Namespace
